src/prefect/server/services/telemetry.py

```diff
@@ -1,143 +1,114 @@
-    LoopService,
-    RunInEphemeralServers,
-    RunInWebservers,
-)
+from prefect.server.services.perpetual_services import perpetual_service
 from prefect.settings import PREFECT_DEBUG_MODE
 from prefect.settings.context import get_current_settings
-from prefect.settings.models.server.services import ServicesBaseSetting
 from prefect.types._datetime import now
 
+logger: logging.Logger = get_logger(__name__)
+
 
-class Telemetry(RunInEphemeralServers, RunInWebservers, LoopService):
+async def _fetch_or_set_telemetry_session(
+    db: PrefectDBInterface,
+) -> tuple[str, str]:
     """
-    Sends anonymous data to Prefect to help us improve
+    Fetch or create a telemetry session in the configuration table.
 
-    It can be toggled off with the PREFECT_SERVER_ANALYTICS_ENABLED setting.
+    Returns:
+        tuple of (session_start_timestamp, session_id)
     """
+    async with db.session_context(begin_transaction=True) as session:
+        telemetry_session = await configuration.read_configuration(
+            session, "TELEMETRY_SESSION"
+        )
 
-    loop_seconds: float = 600
+        if telemetry_session is None:
+            logger.debug("No telemetry session found, setting")
+            session_id = str(uuid4())
+            session_start_timestamp = now("UTC").isoformat()
+
+            telemetry_session = Configuration(
+                key="TELEMETRY_SESSION",
+                value={
+                    "session_id": session_id,
+                    "session_start_timestamp": session_start_timestamp,
+                },
+            )
 
-    @classmethod
-    def service_settings(cls) -> ServicesBaseSetting:
-        raise NotImplementedError("Telemetry service does not have settings")
+            await configuration.write_configuration(session, telemetry_session)
+        else:
+            logger.debug("Session information retrieved from database")
+            session_id = telemetry_session.value["session_id"]
+            session_start_timestamp = telemetry_session.value["session_start_timestamp"]
 
-    @classmethod
-    def environment_variable_name(cls) -> str:
-        return "PREFECT_SERVER_ANALYTICS_ENABLED"
+    logger.debug(f"Telemetry Session: {session_id}, {session_start_timestamp}")
+    return (session_start_timestamp, session_id)
 
-    @classmethod
-    def enabled(cls) -> bool:
-        return get_current_settings().server.analytics_enabled
 
-    def __init__(self, loop_seconds: Optional[int] = None, **kwargs: Any):
-        super().__init__(loop_seconds=loop_seconds, **kwargs)
-        self.telemetry_environment: str = os.environ.get(
-            "PREFECT_API_TELEMETRY_ENVIRONMENT", "production"
-        )
+@perpetual_service(
+    enabled_getter=lambda: get_current_settings().server.analytics_enabled,
+    run_in_ephemeral=True,
+    run_in_webserver=True,
+)
+async def send_telemetry_heartbeat(
+    perpetual: Perpetual = Perpetual(automatic=True, every=timedelta(seconds=600)),
+) -> None:
```

Comment:
How do the `@perpetual_service` config and `automatic=True` interact with each other? Would background service deployments also send heartbeats?

Reply:
i added some context to the PR body, lmk what you think!